3.analyze_data/nbconverted/analyze_predicted_probabilities.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pathlib
+import sys
+
+import numpy as np
+import pandas as pd
+from scikit_posthocs import posthoc_dunn
+from scipy.stats import mannwhitneyu
+
+# Import significance test utils
+sys.path.append("utils")
+import significance_testing as sig_test
+
+
+# ## Find the root of the git repo on the host system
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Input Paths
+
+# In[3]:
+
+
+# Input paths
+big_drive_path = f"{root_dir}/big_drive"
+sqlite_data_path = f"{big_drive_path}/sc_data"
+ref_path = f"{root_dir}/reference_plate_data"
+proba_path = f"{big_drive_path}/probability_sc_data/model_probabilities.parquet"
+sig_test_path = "utils/significance_testing"
+
+# Define the probabilities dataframe
+probadf = pd.read_parquet(proba_path)
+
+# Metadata and platemap paths and the name of the treatment_columns for each treatment type
+treatment_paths = {"compound":
+               {"metadata": pd.read_csv(f"{ref_path}/JUMP-Target-1_compound_metadata_targets.tsv", sep="\t"), "platemap": pd.read_csv(f"{ref_path}/JUMP-Target-1_compound_platemap.txt", sep="\t"), "treatment_column": "pert_iname"},
+               "crispr":
+               {"metadata": pd.read_csv(f"{ref_path}/JUMP-Target-1_crispr_metadata.tsv", sep="\t"), "platemap": pd.read_csv(f"{ref_path}/JUMP-Target-1_crispr_platemap.txt", sep="\t"), "treatment_column": "target_sequence"},
+               "orf":
+               {"metadata": pd.read_csv(f"{ref_path}/JUMP-Target-1_orf_metadata.tsv", sep="\t"), "platemap": pd.read_csv(f"{ref_path}/JUMP-Target-1_orf_platemap.txt", sep="\t"), "treatment_column": "gene"}}
+
+
+# ## Define and create the output paths
+
+# In[4]:
+
+
+filename = "comparison_results.parquet"
+output_path = pathlib.Path(f"{big_drive_path}/statistical_test_comparisons")
+output_path.mkdir(parents=True, exist_ok=True)
+
+
+# In[5]:
+
+
+treatment_paths["compound"]["platemap"]["broad_sample"].fillna("DMSO", inplace=True)
+
+
+# ## Mann-whitney U wrapper function
+
+# In[6]:
+
+
+def perform_mannwhitneyu_median(_dmso_probs, _treatment_probs):
```

Comment:
consider moving functions to top of file

Reply:
I've thought about this, I think they are as far up as possible without displacing other sections that belong at the top. Right now the hierarchy goes: Imports -> input paths -> output paths -> functions -> rest of the code